packages/react-components/src/components/datatree-output-component.tsx

```diff
@@ -154,4 +160,45 @@ export class DataTreeOutputComponent extends AbstractOutputComponent<AbstractOut
         // fix Warning: Can't perform a React state update on an unmounted component
         this.setState = (_state, _callback) => undefined;
     }
+
+    protected async fetchSelectionData(): Promise<void> {
+        if (this.props.selectionRange) {
+            const origXYTree = cloneDeep(this.state.xyTree);
+            origXYTree.sort((a,b)=> (a.id > b.id) ? 1 : -1);
+
+            let payload: any;
+            if (Number(this.props.selectionRange.getStart()) < Number(this.props.selectionRange.getEnd())) {
+                payload = QueryHelper.timeQuery([this.props.selectionRange.getStart(), this.props.selectionRange.getEnd()]);
+            } else {
+                payload = QueryHelper.timeQuery([this.props.selectionRange.getEnd(), this.props.selectionRange.getStart()]);
+            }
+
+            payload.parameters.isFiltered = true;
+
+            // TODO: use the data tree endpoint instead of the xy tree endpoint
+            const tspClientResponse = await this.props.tspClient.fetchXYTree(this.props.traceId, this.props.outputDescriptor.id, payload);
+            const treeResponse = tspClientResponse.getModel();
+            if (tspClientResponse.isOk() && treeResponse) {
+                if (treeResponse.model) {
+                    this.setState({
+                        outputStatus: treeResponse.status,
+                        xyTree: treeResponse.model.entries,
+                    });
+                }
+            }
+        }
+    }
+
+    async componentDidUpdate(prevProps: DataTreeOutputProps): Promise<void> {
+        if (this.props.selectionRange && this.props.selectionRange !== prevProps.selectionRange) {
+            if (Math.abs(Number(this.props.selectionRange.getDuration())) > 0) {
+                this._debouncedFetchSelectionData();
+            }
+            else {
+                this.setState({
+                    xyTree: this._totalTree
```

Comment:
In that case this _totalTree may no longer be needed.

Reply:
@PatrickTasse Updated componentDidUpdate() implementation